test/query.spec.js

```diff
@@ -1,429 +1,271 @@
-            });
-        });
-
-        it ("A query to Vehicle must have result count to 1 w/o reasoning", function(done) {
-
-            conn.query({
-                database: "nodeDBReasoning",
-                query: "select distinct ?s where { ?s a <http://example.org/vehicles/Vehicle> }",
-                limit: 20,
-                offset: 0,
-                reasoning: false
-            },
-            function (data) {
-                // console.log(data);
-
-                expect(data).not.to.be(null);
-                expect(data.results).not.to.be(undefined);
-                expect(data.results.bindings).not.to.be(undefined);
-                expect(data.results.bindings.length).to.be(0);
-                done();
-            });
-        });
-
-    });
-}));
+const Stardog = require('../lib');
+const {
+  seedDatabase,
+  dropDatabase,
+  generateDatabaseName,
+} = require('./setup-database');
+
+describe('query()', () => {
+  const database = generateDatabaseName();
+  let conn;
+
+  beforeAll(seedDatabase(database));
+  afterAll(dropDatabase(database));
+
+  beforeEach(() => {
+    conn = new Stardog.Connection();
+    conn.setEndpoint('http://localhost:5820/');
+    conn.setCredentials('admin', 'admin');
+    conn.setReasoning(true);
+  });
+
+  it('A query result should not be empty', done => {
+    conn.query(
+      {
+        database,
+        query: 'select distinct ?s where { ?s ?p ?o }',
+      },
+      data => {
+        expect(data.results.bindings).toHaveLength(10);
+        done();
+      }
+    );
+  });
+
+  it('A query result should work with property paths', done => {
+    conn.query(
+      {
+        database,
+        query: `select * {
+          ?a a <http://localhost/vocabulary/bench/Article> ; 
+               <http://purl.org/dc/elements/1.1/title> ?title ; 
+               <http://purl.org/dc/elements/1.1/creator> ?c . 
+          ?c (<http://xmlns.com/foaf/0.1/firstName> | <http://xmlns.com/foaf/0.1/lastName>)+ ?name
+        }`,
+      },
+      data => {
+        expect(data.results.bindings).toHaveLength(6);
+        done();
+      }
+    );
+  });
+
+  it('A query result should not have more bindings than its intended limit', done => {
+    conn.query(
+      {
+        database,
+        query: 'select * where { ?s ?p ?o }',
+        limit: 10,
+        offset: 0,
+      },
+      data => {
+        expect(data.results.bindings).toHaveLength(10);
+        done();
+      }
+    );
+  });
+
+  it('The baseURI option should be applied to the query', done => {
+    conn.query(
+      {
+        database,
+        query: 'select * where { <Article1> ?p ?o }',
+        baseURI: 'http://localhost/publications/articles/Journal1/1940/',
+        limit: 10,
+        offset: 0,
+      },
+      data => {
+        expect(data.results.bindings).toHaveLength(10);
+        // console.log(require('util').inspect(data.results, true, null, true))
+        done();
+      }
+    );
+  });
+
```

Comment:
It's basically another way of defining the `:` prefix and doing `select * {?s a :Article}`